src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java

```diff
@@ -98,10 +120,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       // register the the question's general answer statistics
       updateGeneralQuestionStats(questionInGameInstanceDocRef, isAnswerCorrect, firestoreDb);
-
+      if(answerId != null && !answerId.isEmpty()){
```

Comment:
I'm thinking that maybe we should make sure that an answerId cannot be empty when it is sent to the servlet? Or if it is possible, don't count it at all. Because if it is empty, we are already registering that the student got it wrong in his record (call in line 109), but we are not saving this "question answer" in the general statistics for the question, so this could lead to data inconsistency in the DB.